src/commands/download/firefox.ts

```diff
@@ -111,17 +112,23 @@ async function downloadFirefoxSource(version: string) {
 }
 
 export async function downloadInternals(version: string) {
-  // If gFFVersion isn't specified, provide legible error
+  // Provide a legible error if there is no version specified
   if (!version) {
     log.error(
       'You have not specified a version of firefox in your config file. This is required to build a firefox fork.'
     )
     process.exit(1)
   }
 
-  if (existsSync(ENGINE_DIR)) {
-    log.info("Deleting engine/")
-    rmSync(ENGINE_DIR)
+  // If the engine directory is empty, we should delete its contents
+  const engineIsEmpty = await readdir(ENGINE_DIR).then((files) => files.length === 0)
+  if (existsSync(ENGINE_DIR) && engineIsEmpty) {
+    log.info("'engine/' is empty, deleting contents...")
+    rmSync(ENGINE_DIR, { recursive: true })
+
+  }
```

Comment:
What happens if it is empty? I suggest that it outputs something rather than doing nothing.

Reply:
Do you mean if it isn't empty? If firefox has already been downloaded, it will skip downloading it and go to downloading addons.